### PR TITLE
Update to Quarkus 2.2.3 and OpenJDK 17

### LINKED ISSUE #7: Upgrade to Quarkus 2.0
The new dev services should also make the custom Postgres test resource superfluous.

### PR DESCRIPTION
**Pull request by thomasdarimont**
This PR upgrades the example to Quarkus 2.2.3 and OpenJDK 17.

- Adapted code and configuration for quarkus 2.2.3
- Changed jar generation to uber-jar to ease running the app
- Updated dockerfile to work with OpenJDK 17 (Note, there seems to be no java-17-headless package yet...)
- Adjusted docker-compose file to work with newer image versions
- Changed `priority` field in `TodoForm` from int to String to workaround quarkus issue https://github.com/quarkusio/quarkus/issues/19280

This PR depends on the https://github.com/gunnarmorling/jfr-datasource/pull/1 being merged.

Fixes #7.

### CODE REVIEW

File: example-service/pom.xml [hunk 1]
```diff
@@ -8,14 +8,14 @@
...
+    <maven.compiler.target>17</maven.compiler.target>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-    <quarkus-plugin.version>1.6.0.Final</quarkus-plugin.version>
     <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
-    <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
-    <quarkus.platform.version>1.6.0.Final</quarkus.platform.version>
-    <surefire-plugin.version>2.22.1</surefire-plugin.version>
+    <quarkus.platform.group-id>io.quarkus.platform</quarkus.platform.group-id>
+    <quarkus.platform.version>2.2.2.Final</quarkus.platform.version>
```
Comment (gastaldi) [inline example-service/pom.xml:17]:
```suggestion
    <quarkus.platform.version>2.2.3.Final</quarkus.platform.version>
```

Comment (thomasdarimont) [reply example-service/pom.xml]: Just updated Quarkus to 2.2.3.Final

**Comment by gunnarmorling** [comment]
Wow, bit rot is real, it seems. Thanks a lot for all the updates, @thomasdarimont! Moving from MP Metrics to Micrometer probably would be another worthwhile update at some point, as I think the former is being phased out over time.

**Comment by gunnarmorling** [comment]
Sent you an invite as a collaborator too, in case you'd like to push more stuff :)

**Pull request merged**